### PR TITLE
PWX-33942: restart px after expansion

### PR DESCRIPTION
**Pull request by alicelyu-px**
Added new test case "Restart PX after pool expansion" in pool_functional_test.go
Renamed poolToBeResized -> poolToResize

Successful test run at 
https://jenkins.pwx.dev.purestorage.com/job/Users/job/alyu/job/pxdv-7-dev/77/console

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -216,16 +216,47 @@ var _ = Describe("{PoolExpandWithPXRestart}", func() {
...
+			log.FailOnError(Inst().V.WaitDriverUpOnNode(*storageNode, addDriveUpTimeOut),
+				fmt.Sprintf("Timed out waiting for px to come up on node [%s]", storageNode.Name))
+		})
+
+		Step("Ensure pool is up and running", func() {
+			// Ensure pool is up and running
+			poolToResize = getStoragePool(poolIDToResize)
+			// Ensure poolToResize is not nil
+			dash.VerifyFatal(poolToResize != nil, true, "Pool is up and running after restart")
+			verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
```
Comment (pp511) [inline tests/basic/pool_function_test.go:246]: Q. In what cases do we expect new size to be equal to existing size after an expand?

**Review by pp511 — COMMENTED** [review]
Looks good.  Some nits.
Just want you to answer one question. Can approve after that.

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -216,16 +216,47 @@ var _ = Describe("{PoolExpandWithPXRestart}", func() {
...
+			targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+			targetSizeGiB = targetSizeInBytes / units.GiB
+			log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type add-disk",
+				poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+			triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK)
+		})
+
+		Step("Wait for expansion to finish and restart PX", func() {
+			resizeErr := waitForOngoingPoolExpansionToComplete(poolIDToResize)
+			dash.VerifyFatal(resizeErr, nil, "Pool expansion does not result in error")
```
[the remark this answers is not in the document]
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:234]: This line describes the correct/expected behavior, not "what happened", therefore present tense.

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:246]: Not sure I get your question.. Here we are checking the updated size is equal or larger than the "targetSize", not "existing size", wonder if you were confused.

Comment (pp511) [reply tests/basic/pool_function_test.go:246]: Thanks for clarifying.

**Review by pp511 — APPROVED** [review]
lgtm

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -216,16 +216,47 @@ var _ = Describe("{PoolExpandWithPXRestart}", func() {
 		EndTorpedoTest()
 	})
 
+	It("Restart PX after pool expansion", func() {
+		StartTorpedoTest("RestartAfterPoolExpansion",
+			"Restart PX after pool expansion", nil, testrailID)
```
Comment (pp511) [inline tests/basic/pool_function_test.go:221]: @alicelyu-px I missed this in your PR and assumed that this is an existing test so testrailID would be initialized already. I don't see testrailID being initialized here. Can you help me understand what am I missing? If it is something you missed, please fix it.

**Pull request merged**